### PR TITLE
perf(ImageUsage): use PaintImage trace events when available

### LINKED ISSUE #2407: Optimize images don't support cross-domain images
Testing my live URL in Page speed insights gives me a list as long as my arm of images that can be optimized further... 

[https://developers.google.com/speed/pagespeed/insights/?url=http%3A%2F%2Fwww.partydelights.co.uk%2Fthemes%2Fteenage-mutant-ninja-turtles-party-supplies.aspx%3Fpmo%3D3]()

<details><summary>List</summary>
<pre>
> Optimize the following images to reduce their size by 358.9KiB (27% reduction).
> Compressing http://images.partydelights.co.uk/…s/Turtle-Pinata-Kit-PINA062K_v2_ps14.jpg could save 51.1KiB (74% reduction).
> Compressing http://www.partydelights.co.uk/…tant-turtles/header-ninja-turtles_V2.jpg could save 22.3KiB (23% reduction).
> Compressing http://images.partydelights.co.uk/…s/ninja-turtles-bunting-TMNTBUNT_p60.jpg could save 20.9KiB (53% reduction).
> Compressing http://images.partydelights.co.uk/images/banners/PBANTMNTA/PBANTMNTA.jpg could save 14.2KiB (50% reduction).
> Compressing http://images.partydelights.co.uk/…z-Balloon-24-inch-Foil-FOIL2140_ps13.jpg could save 10KiB (28% reduction).
> Compressing http://images.partydelights.co.uk/…urtles-balloon-bouquet-FOILKIT1_ps15.jpg could save 9.8KiB (23% reduction).
> Compressing http://images.partydelights.co.uk/…blue-star-foil-balloon-FOILSTARB_th2.jpg could save 8.8KiB (70% reduction).
> Compressing http://images.partydelights.co.uk/…-Turtles-Scene-Setter-TMNT2SCEN_ps13.jpg could save 8.2KiB (23% reduction).
> Compressing http://images.partydelights.co.uk/…een-star-foil-balloon-FOILSTARKG_th2.jpg could save 7.3KiB (68% reduction).
> Compressing http://images.partydelights.co.uk/…-lilac-star-foil-balloon-FOIL336_th2.jpg could save 7.1KiB (67% reduction).
> Compressing http://images.partydelights.co.uk/…ble-Decorating-Kit-TMNT2TDEC_v2_ps13.jpg could save 7.1KiB (25% reduction).
> Compressing http://images.partydelights.co.uk/…-green-star-foil-balloon-FOIL328_th2.jpg could save 6.9KiB (66% reduction).
> Compressing http://images.partydelights.co.uk/…les/TMNT-Foil-Banner-4m-TMNTBANN_p60.jpg could save 6.8KiB (23% reduction).
> Compressing http://images.partydelights.co.uk/…ges/posters/TMNT-Poster-PSTR100_ps13.jpg could save 6.2KiB (22% reduction).
> Compressing http://images.partydelights.co.uk/…s/Ninja-Turtles-Banner-TMNT2LETT_p60.jpg could save 5.7KiB (25% reduction).
> Compressing http://images.partydelights.co.uk/…s-Hanging-Decorations-TMNT2HANG_ps13.jpg could save 5.6KiB (22% reduction).
> Compressing http://images.partydelights.co.uk/…les-Plastic-Tablecover-TMNT2TABL_p60.jpg could save 4.9KiB (26% reduction).
> Compressing http://images.partydelights.co.uk/…treamer-jumbo-500ft-v2-ROYB2STRE_p60.jpg could save 4.5KiB (23% reduction).
> Compressing http://images.partydelights.co.uk/…yal-blue-hanging-swirls-DECO443_ps13.jpg could save 4.5KiB (21% reduction).
> Compressing http://images.partydelights.co.uk/…upershape-Foil-Balloon-TMNTSUPE_ps13.jpg could save 4.4KiB (19% reduction).
> Compressing http://images.partydelights.co.uk/…inja-turtles-balloon-AIRW027_V2_ps15.jpg could save 4.4KiB (15% reduction).
> Compressing http://images.partydelights.co.uk/…a-Turtles-Party-Bag-Kit-TMNTBAGK_th2.jpg could save 4.3KiB (27% reduction).
> Compressing http://images.partydelights.co.uk/images/sweets/SWEE122/SWEE122_th2.jpg could save 4.1KiB (29% reduction).
> Compressing http://images.partydelights.co.uk/…ions/Kiwi-Tassel-Garland-DECO695_p60.jpg could save 3.9KiB (20% reduction).
> Compressing http://images.partydelights.co.uk/…Turtles-Crepe-Streamer-TMNT2STRE_p60.jpg could save 3.7KiB (25% reduction).
> Compressing http://images.partydelights.co.uk/…les-Stand-In-Photo-Prop-CUTO305_ps14.jpg could save 3.6KiB (17% reduction).
> Compressing http://images.partydelights.co.uk/…Ninja-Turtles-Confetti-TMNT2CONF_th2.jpg could save 3.6KiB (20% reduction).
> Compressing http://images.partydelights.co.uk/…azaroo-Orange-Face-Paint-FACE025_th2.jpg could save 3.2KiB (27% reduction).
> Compressing http://images.partydelights.co.uk/…bright-red-face-paint-face007_v3_th2.jpg could save 3KiB (25% reduction).
> Compressing http://images.partydelights.co.uk/images/balloons/TMNTMINI_th2.jpg could save 2.9KiB (30% reduction).
> Compressing http://images.partydelights.co.uk/…tles-Party-Invitations-TMNT2INVI_th2.jpg could save 2.9KiB (24% reduction).
> Compressing http://images.partydelights.co.uk/…l/three-tier-cake-stand-DOTSSTAN_th2.jpg could save 2.7KiB (27% reduction).
> Compressing http://images.partydelights.co.uk/…urtles-Party-Thank-You-TMNT2THAN_th2.jpg could save 2.7KiB (24% reduction).
> Compressing http://images.partydelights.co.uk/…les/Ninja-Turtles-Cups-TMNT2CUPS_th2.jpg could save 2.7KiB (26% reduction).
> Compressing http://images.partydelights.co.uk/…nja-Turtles-Party-Bags-TMNT2LOOT_th2.jpg could save 2.6KiB (24% reduction).
> Compressing http://images.partydelights.co.uk/…nja-Turtles-Party-Game-TMNT2GAME_th2.jpg could save 2.6KiB (24% reduction).
> Compressing http://images.partydelights.co.uk/…oons/ninja-turtle-balloon-BB137_ps13.jpg could save 2.6KiB (14% reduction).
> Compressing http://images.partydelights.co.uk/…Ninja-Turtles-Gift-Cup-TMNT2FCUP_th2.jpg could save 2.5KiB (25% reduction).
> Compressing http://images.partydelights.co.uk/…ja-Turtles-Plates-23cm-TMNT2PLAT_th2.jpg could save 2.4KiB (19% reduction).
> Compressing http://images.partydelights.co.uk/…-Turtles-Balloon-Weight-TMNTWEIG_th2.jpg could save 2.4KiB (22% reduction).
> Compressing http://images.partydelights.co.uk/…-Bright-Green-Face-Paint-FACE005_th2.jpg could save 2.4KiB (22% reduction).
> Compressing http://images.partydelights.co.uk/…a-Turtles-Paddle-Balls-TMNT2PADD_th2.jpg could save 2.4KiB (25% reduction).
> Compressing http://images.partydelights.co.uk/…ja-Turtles-Plates-17cm-TMNT2DESS_th2.jpg could save 2.3KiB (19% reduction).
> Compressing http://images.partydelights.co.uk/…/Ninja-Turtles-Napkins-TMNT2NAPK_th2.jpg could save 2.3KiB (20% reduction).
> Compressing http://images.partydelights.co.uk/…tles/TMNT-Foil-Balloon-TMNTFOIL2_th2.jpg could save 2.3KiB (19% reduction).
> Compressing http://images.partydelights.co.uk/…lue-royal-blue-and-red-TISS85_V2_th2.jpg could save 2.3KiB (22% reduction).
> Compressing http://images.partydelights.co.uk/…-Turtles-Favour-Bucket-TMNT2BUCK_th2.jpg could save 2.2KiB (24% reduction).
> Compressing http://images.partydelights.co.uk/…urtles-Beverage-Napkins-TMNT2BEV_th2.jpg could save 2.2KiB (23% reduction).
> Compressing http://images.partydelights.co.uk/…inatas/turtle-pinata-PINA062_v2_ps14.jpg could save 2.1KiB (18% reduction).
> Compressing http://images.partydelights.co.uk/…oo-Royal-Blue-Face-Paint-FACE032_th2.jpg could save 2.1KiB (19% reduction).
> Compressing http://images.partydelights.co.uk/…angelo-Cardboard-Cutout-CUTO302_ps14.jpg could save 2.1KiB (16% reduction).
> Compressing http://images.partydelights.co.uk/…TMNT-11-Latex-Balloons-TMNTBALL2_th2.jpg could save 2KiB (21% reduction).
> Compressing http://images.partydelights.co.uk/…onardo-Cardboard-Cutout-CUTO304_ps14.jpg could save 2KiB (15% reduction).
> Compressing http://images.partydelights.co.uk/…atello-Cardboard-Cutout-CUTO301_ps14.jpg could save 2KiB (15% reduction).
> Compressing http://images.partydelights.co.uk/…les/TMNT-Sticker-Strips-TMNTSTIC_th2.jpg could save 1.9KiB (18% reduction).
> Compressing http://images.partydelights.co.uk/…sonalised-confetti-CONPERSTRA_v2_co2.jpg could save 1.9KiB (19% reduction).
> Compressing http://images.partydelights.co.uk/…sonalised-confetti-CONPERSTRA_v2_co2.jpg could save 1.9KiB (19% reduction).
> Compressing http://images.partydelights.co.uk/…nja-Turtles-Card-Masks-TMNT2MASK_th2.jpg could save 1.8KiB (20% reduction).
> Compressing http://images.partydelights.co.uk/…ja-Turtles-Sweat-Bands-TMNT2SBAN_th2.jpg could save 1.8KiB (25% reduction).
> Compressing http://images.partydelights.co.uk/images/toys/TOYS035_th2.jpg could save 1.8KiB (25% reduction).
> Compressing http://images.partydelights.co.uk/…inja-Turtles-Party-Masks-MASK713_th2.jpg could save 1.8KiB (19% reduction).
> Compressing http://images.partydelights.co.uk/…nja-Turtles-Sunglasses-TMNT2GLAS_th2.jpg could save 1.7KiB (22% reduction).
> Compressing http://images.partydelights.co.uk/…rtles-Party-Bubbles-60ml-BUBB073_th2.jpg could save 1.7KiB (23% reduction).
> Compressing http://images.partydelights.co.uk/…aphael-Cardboard-Cutout-CUTO303_ps14.jpg could save 1.7KiB (13% reduction).
> Compressing http://images.partydelights.co.uk/…a-Turtles-Award-Ribbon-TMNT2AWAR_th2.jpg could save 1.7KiB (23% reduction).
> Compressing http://images.partydelights.co.uk/…-turtles/TMNT-Cone-Hats-TMNTHATS_th2.jpg could save 1.7KiB (25% reduction).
> Compressing http://images.partydelights.co.uk/…/ninja-turtles-tumbler-TMNTCUPS2_th2.jpg could save 1.6KiB (24% reduction).
> Compressing http://images.partydelights.co.uk/…azaroo-Purple-Face-Paint-FACE031_th2.jpg could save 1.6KiB (16% reduction).
> Compressing http://images.partydelights.co.uk/…es/ninja-turtles-badges-BADGE623_th2.jpg could save 1.6KiB (19% reduction).
> Compressing http://images.partydelights.co.uk/…-Balloon-30-inch-Foil-TMNTSUPE2_ps13.jpg could save 1.6KiB (14% reduction).
> Compressing http://images.partydelights.co.uk/…ibo-terrific-turtles-tub-SWEE297_th2.jpg could save 1.6KiB (18% reduction).
> Compressing http://images.partydelights.co.uk/…eady-to-roll-icing-green-SDEC209_th2.jpg could save 1.5KiB (22% reduction).
> Compressing http://images.partydelights.co.uk/…es-Balloon-12-inch-Foil-TMNTAIRF_th2.jpg could save 1.4KiB (24% reduction).
> Compressing http://images.partydelights.co.uk/…ions/green-hanging-fans-DECO439_ps13.jpg could save 1.4KiB (13% reduction).
> Compressing http://images.partydelights.co.uk/…urtles-plastic-cup-straw-CATE567_th2.jpg could save 1.3KiB (22% reduction).
> Compressing http://images.partydelights.co.uk/…/Ninja-Turtles-Pencils-TMNT2PENC_th2.jpg could save 1.3KiB (19% reduction).
> Compressing http://images.partydelights.co.uk/…il/purple-foil-weight-WEIG144_V2_th2.jpg could save 1.2KiB (14% reduction).
> Compressing http://images.partydelights.co.uk/…-turtles/TMNT-Straws-TMNTSTRA_v2_th2.jpg could save 1.1KiB (19% reduction).
> Compressing http://images.partydelights.co.uk/…s/TMNT-Rubber-Bracelets-TMNTBRAC_th2.jpg could save 1KiB (17% reduction).
> Compressing http://images.partydelights.co.uk/…x-balloon-11-inch-green-ball1147_th2.jpg could save 1KiB (20% reduction).
> Compressing http://images.partydelights.co.uk/…ime-Green-Plastic-Bucket-BOXF196_th2.jpg could save 1,015B (22% reduction).
> Compressing http://images.partydelights.co.uk/…s/sparkling-sugar-purple-SDEC179_th2.jpg could save 932B (14% reduction).
> Compressing http://images.partydelights.co.uk/…lloon-11-inch-royalblue-ball1145_th2.jpg could save 886B (16% reduction).
> Compressing http://images.partydelights.co.uk/…lloon-11-inch-limegreen-ball1137_th2.jpg could save 869B (22% reduction).
> Compressing http://images.partydelights.co.uk/images/balloons/CURL500EME_th2.jpg could save 866B (16% reduction).
> Compressing http://images.partydelights.co.uk/…t-turtles/TMNT-Blowouts-TMNTBLOW_th2.jpg could save 796B (16% reduction).
> Compressing http://images.partydelights.co.uk/…andy-buffet/Kiwi-Scoop-CBUFSCOO4_th2.jpg could save 767B (21% reduction).
> Compressing http://images.partydelights.co.uk/images/balloons/CURL500LAV_th2.jpg could save 745B (15% reduction).
> Compressing http://images.partydelights.co.uk/images/catering_rolls/CATE045_th2.jpg could save 634B (16% reduction).
> Compressing http://images.partydelights.co.uk/…-balloon-11-inch-purple-ball1139_th2.jpg could save 604B (14% reduction).
> Compressing http://images.partydelights.co.uk/…/plain/royal-blue-bowl-ROYB2BOWL_th2.jpg could save 563B (16% reduction).
</pre>
</details>

but Lighthouse is only reporting one of these images as needing attention...

> …mutant-turtles/header-ninja-turtles_V2.jpg
> 100 KB
> 14 KB (14%)

Should the results be the same as in page speed insights? 
I was hoping to use Lighthouse for this test before my images get published.

### PR DESCRIPTION
**Pull request by patrickhulce**
eliminates the need to execute the in page listener and re-evaluate the image for natural size data if paintimage data is available in the trace events

fixes #2407

### CODE REVIEW

**Comment by somemy** [comment]
```
#Conflicting files
#`lighthouse-core/audits/byte-efficiency/uses-responsive-images.js`
#[lighthouse-core/gather/gatherers/image-usage.js]([url](url))

```

**Comment by somemy** [comment]
lighthouse/lighthouse-core/test/gather/gatherers/image-usage-test.js

File: lighthouse-cli/test/fixtures/byte-efficiency/tester.html [hunk 1]
```diff
@@ -129,9 +129,9 @@ <h2>Byte efficiency tester page</h2>
   <!-- PASS(responsive): image is used numerous times in sprite-fashion -->
   <!-- PASS(offscreen): image is onscreen -->
   <div class="onscreen" style="width: 30px; height: 30px; background: 0% 50% url(lighthouse-480x320.jpg?sprite);"></div>
-  <div class="onscreen" style="width: 30px; height: 30px; background: 25% 50% url(lighthouse-480x320.jpg?sprite);"></div>
-  <div class="onscreen" style="width: 30px; height: 30px; background: 50% 50% url(lighthouse-480x320.jpg?sprite);"></div>
-  <div class="onscreen" style="width: 30px; height: 30px; background: 75% 50% url(lighthouse-480x320.jpg?sprite);"></div>
+  <div class="onscreen" style="left: 10px; width: 30px; height: 30px; background: 25% 50% url(lighthouse-480x320.jpg?sprite);"></div>
+  <div class="onscreen" style="left: 20px; width: 30px; height: 30px; background: 50% 50% url(lighthouse-480x320.jpg?sprite);"></div>
+  <div class="onscreen" style="left: 30px; width: 30px; height: 30px; background: 75% 50% url(lighthouse-480x320.jpg?sprite);"></div>
 </div>
```
Comment (somemy) [inline lighthouse-cli/test/fixtures/byte-efficiency/tester.html]: lighthouse/lighthouse-core/test/gather/gatherers/image-usage-test.js

Comment (somemy) [reply lighthouse-cli/test/fixtures/byte-efficiency/tester.html]: https://web.facebook.com/mewkungza.bkk?ref=br_rs

Comment (somemy) [reply lighthouse-cli/test/fixtures/byte-efficiency/tester.html]: https://guides.github.com/feed/index.xml

**Comment by somemy** [comment]
https://github.com/marionebl/commitlint/blob/master/docs/reference-rules.md

File: lighthouse-cli/test/fixtures/byte-efficiency/tester.html (hunk 1 above)
Comment (somemy) [reply lighthouse-cli/test/fixtures/byte-efficiency/tester.html]: https://github.com/marionebl/commitlint/blob/master/docs/reference-rules.md

File: lighthouse-core/audits/byte-efficiency/offscreen-images.js [hunk 2]
```diff
@@ -59,8 +59,14 @@ class OffscreenImages extends ByteEfficiencyAudit {
    */
```
Comment (somemy) [inline lighthouse-core/audits/byte-efficiency/offscreen-images.js]: https://github.com/marionebl/commitlint/blob/master/docs/reference-rules.md

**Comment by somemy** [comment]
/html/body/pre/text()[1]

File: lighthouse-cli/test/fixtures/byte-efficiency/tester.html [hunk 3]
```diff
@@ -129,9 +129,9 @@ <h2>Byte efficiency tester page</h2>
   <!-- PASS(responsive): image is used numerous times in sprite-fashion -->
```
Comment (somemy) [inline lighthouse-cli/test/fixtures/byte-efficiency/tester.html]: http://www.apache.org/licenses/

**Comment by somemy** [comment]
https://developers.google.com/open-source/github/accounts#googlebot

**Comment by brendankenny** [comment]
so... @patrickhulce, what do you think? :)

**Comment by patrickhulce** [comment]
meh, we added a few more image features since this was written so we couldn't always use this. I think it's safe to abandon. If we need to squeeze more perf out of ImageUsage we can always come back to this.

**Pull request closed without merging**